### PR TITLE
Fix #25

### LINKED ISSUE #25: [1.7.10] Blacked GUI sometimes
Bug from https://bitbucket.org/binnie567/binnie-mods/issues/327/genepool-gui-rawdna-and-power-black-out

Original text:

> The RawDNA and Power GUI bars go black when the genepool is in use. Hovering mouse cursor over the RawDNA tank renders both back to normal but as soon as you move off the RawDNA tank they go black again.

### PR DESCRIPTION
**Pull request by KorDum**
Fix #25
Fix wrong OpenGL in custom GUI
Some refactoring

### CODE REVIEW

**Comment by KorDum** [comment]
Please code review

**Review by marcin212 — APPROVED** [review]

**Review by Nedelosk — APPROVED** [review]

File: src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java [hunk 1]
```diff
@@ -332,86 +338,84 @@ public void renderTexturePadded(IArea area, IArea texture, IBorder padding) {
...
 	}
 
 	public void renderItem(IPoint pos, ItemStack item, boolean rotating) {
-		renderItem(pos, item, (int) zLevel + 3, rotating);
-	}
-
-	private void renderItem(IPoint pos, ItemStack item, int zLevel, boolean rotating) {
 		OpenGlHelper.setLightmapTextureCoords(OpenGlHelper.lightmapTexUnit, 240.0f, 240.0f);
-		GL11.glPushMatrix();
+		GL11.glPushAttrib(8256);
```
Comment (mezz) [inline src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java:374]: What is this change?

**Review by mezz — COMMENTED** [review]
Please split up cleanup from the changes before I can review properly. This is much harder to review this way.

File: src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java (hunk 1 above)
Comment (KorDum) [reply src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java:374]:
```
public static void pushAttrib()
    {
        GL11.glPushAttrib(8256);
    }
```
in GlStateManager from Forge 1.11.2.
https://github.com/ForestryMC/Binnie/blob/master-MC1.11.2/src/main/java/binnie/core/craftgui/renderer/RenderUtil.java#L45

**Comment by KorDum** [comment]
I splitted changes.

**Comment by mezz** [comment]
Looks good to me. Knowing OpenGL issues, this may break something obscure in another place, but it does look like an improvement.

**Comment by KorDum** [comment]
Can I merge it?

**Comment by mezz** [comment]
Go ahead. 
Should I make a new build after this is merged?

**Comment by KorDum** [comment]
> Should I make a new build after this is merged?

Yeah!

**Comment by mezz** [comment]
Ok submitted here: https://minecraft.curseforge.com/projects/binnies-mods/files/2444725

**Comment by KorDum** [comment]
Thanks!

**Pull request merged**